tw-tasks-core/src/test/java/com/transferwise/tasks/helpers/kafka/partitionkey/RandomPartitionKeyStrategyTest.java

```diff
@@ -0,0 +1,36 @@
+package com.transferwise.tasks.helpers.kafka.partitionkey;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.transferwise.tasks.domain.BaseTask;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RandomPartitionKeyStrategyTest {
+
+  private RandomPartitionKeyStrategy subject;
+
+  @BeforeEach
+  void setUp() {
+    subject = new RandomPartitionKeyStrategy();
+  }
+
+  @Test
+  void shouldAlwaysReturnADifferentKey() {
+    final var id = UUID.randomUUID();
+    final var baseTask = new BaseTask()
+        .setId(id)
+        .setType("type")
+        .setPriority(1)
+        .setVersion(1L);
+
+    final var keys = IntStream.range(0, 10)
+        .mapToObj(i -> subject.createPartitionKey(baseTask))
+        .collect(Collectors.toList());
+
+    keys.forEach(k -> assertThat(keys).containsOnlyOnce(k));
```

Comment:
This should have ~ 99.931% success rate.
65536*65535*...*65527 / 65536 ^ 10 .

Would it be good enough?

Maybe we could run it twice/trice and if one succeeds we are happy?

Reply:
Refactored the test a bit